newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionParametersEditor.js

```diff
@@ -348,6 +428,12 @@ export default class EventsFunctionParametersEditor extends React.Component<
                               />
                             )}
                           </ResponsiveLineStackLayout>
+                          {parameter.getType() === 'stringWithSelector' && (
+                            <StringSelectorEditor
+                              extraInfo={parameter.getExtraInfo()}
+                              setExtraInfo={parameter.setExtraInfo}
```

Comment:
```suggestion
                              extraInfo={parameter.getExtraInfo()}
                              setExtraInfo={(extraInfo) => { parameter.setExtraInfo(extraInfo); forceUpdate(); }
```

why putting the forceUpdate here? Because the component "StringSelectorEditor" just cares about two things:
- it shows the content of extraInfo
- it tells you when the content of extraInfo changed, giving you the string to store.

so that's what we're doing here, we store the string: `parameter.setExtraInfo(extraInfo)`. However, this thing is stored in the project (i.e: webassembly memory, but it could be anything else: a JS object...). React does not know that `parameter.getExtraInfo()` has now a new value, because it's not a state.
So we call `forceUpdate()` here.

Otherwise, we would not pass the new `extraInfo` to the component (because the render function would not be launched again), which can creates weird behavior and makes the component harder to understand (because, as you did, you need a state inside the component).